integrations/dd-sdk-android-okhttp-otel/src/main/kotlin/com/datadog/android/okhttp/otel/OkHttpExt.kt

```diff
@@ -18,7 +18,9 @@ import okhttp3.Request
  */
 fun Request.Builder.addParentSpan(span: Span): Request.Builder {
     val context = span.spanContext
-    val prioritySampling = if (context.isSampled) PrioritySampling.USER_KEEP.toInt() else PrioritySampling.UNSET.toInt()
+    // because the DatadogInterceptor is using the old legacy DDTracer for handling the spans around request
+    // we need to provide the legacy sampling codes here.
+    val prioritySampling = if (context.isSampled) PrioritySampling.USER_KEEP else PrioritySampling.UNSET
```

Comment:
so the difference is only in the `UNSET` value? `0x80` vs `Integer.MIN_VALUE`. Do we have a possibility of such mismatch in other parts of the code?

Reply:
yes that was the problem, we do not have any other places this is just because we are using Otel here and on the other side we are using OpenTracing. We really need to get rid of that ASAP and switch everything to CoreTracer